xwiki-platform-core/xwiki-platform-web/xwiki-platform-web-war/src/main/webapp/resources/js/xwiki/editors/dataeditors.js

```diff
@@ -17,825 +17,754 @@
-      if (item._expandedBeforeDrag) {
-        item.removeClassName('collapsed');
-      }
-    });
-  }
-});
-
-function init() {
-  require(['scriptaculous/dragdrop'], function() {
-    new editors.XDataEditors()
   });
-  return true;
-}
-
-// When the document is loaded, create the Autosave control
-(XWiki.domIsLoaded && init())
-|| document.observe('xwiki:dom:loaded', init);
-
-// End XWiki augmentation.
-return XWiki;
-}(XWiki || {}));
 
-// Class Switcher
-require(['jquery', 'xwiki-events-bridge'], function($) {
-  $('#switch-xclass').on('change', function(event) {
-    var selectedClass = $(event.target).val();
-    if (selectedClass) {
-      var selectedClassReference = XWiki.Model.resolve(selectedClass, XWiki.EntityType.DOCUMENT,
-        XWiki.currentDocument.documentReference);
-      var selectedClassURL = new XWiki.Document(selectedClassReference).getURL('edit', 'editor=class');
-      var switchClass = function() {
-        window.self.location = selectedClassURL;
-      };
-      new XWiki.widgets.ConfirmationBox({
-        onYes: function() {
-          // Save the current class before switching.
-          $(document).trigger('xwiki:actions:save', {
-            'continue': true,
-            'form': $('#propupdate')[0]
+    let initSwitchClassListener = function () {
+      $('#switch-xclass').on('change', function (event) {
+        let selectedClass = $(event.target).val();
+        if (selectedClass) {
+          let selectedClassReference = XWiki.Model.resolve(selectedClass, XWiki.EntityType.DOCUMENT,
+              XWiki.currentDocument.documentReference);
+          let selectedClassURL = new XWiki.Document(selectedClassReference).getURL(xm.action, 'editor=class');
+          let switchClass = function () {
+            window.self.location = selectedClassURL;
+          };
+          new XWiki.widgets.ConfirmationBox({
+            onYes: function () {
+              // Save the current class before switching.
+              $(document).trigger('xwiki:actions:save', {
+                'continue': true,
+                'form': $('#propupdate')[0]
+              });
+              $(document).on('xwiki:document:saved', switchClass);
+            },
+            // Switch without saving the current class.
+            onNo: switchClass
+          }, {
+            confirmationText: l10n['core.editors.class.switchClass.confirm'],
+            // Allow the users to cancel the switch.
+            showCancelButton: true
           });
-          $(document).on('xwiki:document:saved', switchClass);
-        },
-        // Switch without saving the current class.
-        onNo: switchClass
-      }, {
-        confirmationText: "$services.localization.render('core.editors.class.switchClass.confirm')",
-        // Allow the users to cancel the switch.
-        showCancelButton: true
+        }
       });
     }
+
+    function init() {
+      ((XWiki || {}).editors || {}).XDataEditors = new XDataEditors();
```

Comment:
This looks a bit strange. Don't we want to initialize the missing `XWiki` or `editors`?

Reply:
Hmmm honestly I don't recall exactly why I wrote it like that... but I'm not entirely sure what I should call?

Reply:
I was thinking of:

```
XWiki = window.XWiki || {};
XWiki.editors = XWiki.editors || {};
XWiki.editors.XDataEditors = new XDataEditors();
```